thunder/distributed/__init__.py

```diff
@@ -349,6 +355,176 @@ def f(tensor: TensorProxy) -> str:
     return f
 
 
+# When the user calls fsdp(jitted_module), this function does the following
+# - It transforms the ThunderModule jitted_module, materializing and sharding the parameters as `overrides`
+#   in the ThunderModule.
+# - While doing that, it leaves the original user module alone.
+# - It then registers an early transform (callback that runs before prologue is executed) that transforms the
+#   prologue and compute trace.
+#
+# Note that for doing so, there are a few constraints / caveats:
+# - We do not have prologues/compute traces when we transform the module.
+# - We need to record the info from the module transformations because a later transform might modify the module further.
```

Comment:
could you update the docstring of `fsdp` below?
I'd like to some explanation e.g. when the actual sharding happens